<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CPUView</class>
 <widget class="QMainWindow" name="CPUView">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1058</width>
    <height>673</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>CPU's Debugger</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../resources/appResources.qrc">
    <normaloff>:/resources/Logo.png</normaloff>:/resources/Logo.png</iconset>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_3">
    <item row="0" column="0" rowspan="3">
     <widget class="QTableView" name="disassembly">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>1</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="selectionMode">
       <enum>QAbstractItemView::ExtendedSelection</enum>
      </property>
      <property name="selectionBehavior">
       <enum>QAbstractItemView::SelectRows</enum>
      </property>
      <property name="showGrid">
       <bool>false</bool>
      </property>
      <property name="gridStyle">
       <enum>Qt::NoPen</enum>
      </property>
      <attribute name="horizontalHeaderVisible">
       <bool>false</bool>
      </attribute>
      <attribute name="horizontalHeaderHighlightSections">
       <bool>false</bool>
      </attribute>
     </widget>
    </item>
    <item row="0" column="1">
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="1" column="0">
       <widget class="QTableView" name="stackView">
        <attribute name="horizontalHeaderVisible">
         <bool>false</bool>
        </attribute>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Stack Viewer</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="0" column="2" rowspan="2">
     <layout class="QFormLayout" name="formLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="accumulatorLabel">
        <property name="text">
         <string>Accumulator</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="accumulatorLineEdit">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="programBankRegisterLabel">
        <property name="text">
         <string>Program Bank</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="programBankRegisterLineEdit"/>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="programCounterLabel">
        <property name="text">
         <string>Program Counter</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLineEdit" name="programCounterLineEdit"/>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="directBankLabel">
        <property name="text">
         <string>Direct Bank</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLineEdit" name="directBankLineEdit"/>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="directPageLabel">
        <property name="text">
         <string>Direct Page</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QLineEdit" name="directPageLineEdit"/>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="stackPointerLabel">
        <property name="text">
         <string>Stack Pointer</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QLineEdit" name="stackPointerLineEdit"/>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="xIndexLabel">
        <property name="text">
         <string>X Index</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QLineEdit" name="xIndexLineEdit"/>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="yIndexLabel">
        <property name="text">
         <string>Y Index</string>
        </property>
       </widget>
      </item>
      <item row="7" column="1">
       <widget class="QLineEdit" name="yIndexLineEdit"/>
      </item>
      <item row="8" column="0">
       <widget class="QLabel" name="emulationModeLabel">
        <property name="text">
         <string>Emulation mode</string>
        </property>
       </widget>
      </item>
      <item row="8" column="1">
       <widget class="QCheckBox" name="emulationModeCheckBox">
        <property name="layoutDirection">
         <enum>Qt::RightToLeft</enum>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="1" rowspan="2">
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="loggerLabel">
        <property name="text">
         <string>Instructions History</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QPushButton" name="clear">
        <property name="text">
         <string>Clear History</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QTableView" name="history"/>
      </item>
     </layout>
    </item>
    <item row="2" column="2">
     <widget class="QGroupBox" name="formGroupBox">
      <property name="title">
       <string>Flags</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
      </property>
      <property name="flat">
       <bool>false</bool>
      </property>
      <property name="checkable">
       <bool>false</bool>
      </property>
      <layout class="QFormLayout" name="formLayout_2">
       <property name="sizeConstraint">
        <enum>QLayout::SetDefaultConstraint</enum>
       </property>
       <property name="labelAlignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
       </property>
       <property name="formAlignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
       </property>
       <property name="leftMargin">
        <number>26</number>
       </property>
       <property name="topMargin">
        <number>7</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <item row="0" column="0">
        <widget class="QLabel" name="negativeLabel">
         <property name="text">
          <string>Memory Select (M)</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QCheckBox" name="mCheckbox">
         <property name="layoutDirection">
          <enum>Qt::RightToLeft</enum>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="zeroLabel">
         <property name="text">
          <string>Index Select (X)</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QCheckBox" name="xCheckbox">
         <property name="layoutDirection">
          <enum>Qt::RightToLeft</enum>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="carryLabel">
         <property name="text">
          <string>Interupt Request Disable (I)</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QCheckBox" name="iCheckbox">
         <property name="layoutDirection">
          <enum>Qt::RightToLeft</enum>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="Overflow">
         <property name="text">
          <string>Overflow (V)</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QCheckBox" name="vCheckbox">
         <property name="layoutDirection">
          <enum>Qt::RightToLeft</enum>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QLabel" name="decimalLabel">
         <property name="text">
          <string>Decimal (D)</string>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <widget class="QCheckBox" name="dCheckbox">
         <property name="layoutDirection">
          <enum>Qt::RightToLeft</enum>
         </property>
        </widget>
       </item>
       <item row="6" column="0">
        <widget class="QLabel" name="memoryAccumulatorSelectLabel">
         <property name="text">
          <string>Carry (C)</string>
         </property>
        </widget>
       </item>
       <item row="6" column="1">
        <widget class="QCheckBox" name="cCheckbox">
         <property name="layoutDirection">
          <enum>Qt::RightToLeft</enum>
         </property>
        </widget>
       </item>
       <item row="7" column="0">
        <widget class="QLabel" name="indeXSelectLabel">
         <property name="text">
          <string>Zero (Z)</string>
         </property>
        </widget>
       </item>
       <item row="8" column="0">
        <widget class="QLabel" name="irqDisableLabel">
         <property name="text">
          <string>Negative (N)</string>
         </property>
        </widget>
       </item>
       <item row="8" column="1">
        <widget class="QCheckBox" name="nCheckbox">
         <property name="layoutDirection">
          <enum>Qt::RightToLeft</enum>
         </property>
        </widget>
       </item>
       <item row="7" column="1">
        <widget class="QCheckBox" name="zCheckbox">
         <property name="layoutDirection">
          <enum>Qt::RightToLeft</enum>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="breakBLabel">
         <property name="text">
          <string>Break (B)</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QCheckBox" name="bCheckbox">
         <property name="layoutDirection">
          <enum>Qt::RightToLeft</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="movable">
    <bool>false</bool>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextBesideIcon</enum>
   </property>
   <property name="floatable">
    <bool>true</bool>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionPause"/>
   <addaction name="actionNext"/>
   <addaction name="actionStep"/>
   <addaction name="actionBreakpoint"/>
  </widget>
  <action name="actionPause">
   <property name="icon">
    <iconset resource="../resources/appResources.qrc">
     <normaloff>:/resources/icons/play.svg</normaloff>:/resources/icons/play.svg</iconset>
   </property>
   <property name="text">
    <string>Continue</string>
   </property>
   <property name="toolTip">
    <string>Pause or Resume instruction execution.</string>
   </property>
   <property name="shortcut">
    <string>C</string>
   </property>
  </action>
  <action name="actionStep">
   <property name="icon">
    <iconset resource="../resources/appResources.qrc">
     <normaloff>:/resources/icons/step.svg</normaloff>:/resources/icons/step.svg</iconset>
   </property>
   <property name="text">
    <string>Step</string>
   </property>
   <property name="toolTip">
    <string>Execute a single instruction</string>
   </property>
   <property name="shortcut">
    <string>S</string>
   </property>
  </action>
  <action name="actionNext">
   <property name="icon">
    <iconset resource="../resources/appResources.qrc">
     <normaloff>:/resources/icons/continue.svg</normaloff>:/resources/icons/continue.svg</iconset>
   </property>
   <property name="text">
    <string>Next</string>
   </property>
   <property name="toolTip">
    <string>Continue execution to the next line.</string>
   </property>
   <property name="shortcut">
    <string>N</string>
   </property>
  </action>
  <action name="actionBreakpoint">
   <property name="icon">
    <iconset resource="../resources/appResources.qrc">
     <normaloff>:/resources/icons/breakpoint.svg</normaloff>:/resources/icons/breakpoint.svg</iconset>
   </property>
   <property name="text">
    <string>Breakpoint</string>
   </property>
   <property name="toolTip">
    <string>Create a new custom breakpoint</string>
   </property>
   <property name="shortcut">
    <string>B</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="../resources/appResources.qrc"/>
 </resources>
 <connections/>
</ui>
